include/range/v3/view/drop_while.hpp

```diff
@@ -43,43 +43,22 @@ namespace ranges
             friend range_access;
             using difference_type_ = range_difference_t<Rng>;
             Rng rng_;
-            semiregular_t<function_type<Pred>> pred_;
-            optional<range_iterator_t<Rng>> begin_;
-
-            range_iterator_t<Rng> get_begin_()
-            {
-                if(!begin_)
-                    begin_ = find_if_not(rng_, std::ref(pred_));
-                return *begin_;
-            }
+            variant<semiregular_t<function_type<Pred>>, range_iterator_t<Rng>> begin_;
         public:
             drop_while_view() = default;
-            drop_while_view(drop_while_view &&that)
-              : rng_(std::move(that).rng_), pred_(std::move(that).pred_), begin_{}
-            {}
-            drop_while_view(drop_while_view const &that)
-              : rng_(that.rng_), pred_(that.pred_), begin_{}
-            {}
             drop_while_view(Rng rng, Pred pred)
-              : rng_(std::move(rng)), pred_(as_function(std::move(pred))), begin_{}
+              : rng_(std::move(rng))
+              , begin_{emplaced_index<0>, as_function(std::move(pred))}
             {}
-            drop_while_view& operator=(drop_while_view &&that)
-            {
-                rng_ = std::move(that).rng_;
-                pred_ = std::move(that).pred_;
-                begin_.reset();
-                return *this;
-            }
-            drop_while_view& operator=(drop_while_view const &that)
-            {
-                rng_ = that.rng_;
-                pred_ = that.pred_;
-                begin_.reset();
-                return *this;
-            }
```

Comment:
The point of the copy/assign overload was that if we call `begin` on a `drop_while` range, we end up with an iterator in `begin_`. Then if we copy the range, the destination gets an iterator that points into the underlying source range, not the underlying destination range. So I think we need to keep the code as it was.